lib/ts/types.ts

```diff
@@ -29,10 +29,9 @@ export type AppInfo = {
 
 export type NormalisedAppinfo = {
     appName: string;
-    websiteDomain: NormalisedURLDomain;
```

Comment:
Add an extra field which keeps track of if the original origin passed was a string or a function